runtime/parser/declaration.go

```diff
@@ -339,32 +349,81 @@ func parseAccess(p *parser) (ast.Access, error) {
 			)
 		}
 
-		var access ast.Access
+		var access ast.Access = ast.AccessNotSpecified
 
 		keyword := p.currentTokenSource()
 		switch string(keyword) {
 		case KeywordAll:
 			access = ast.AccessPublic
+			// Skip the keyword
+			p.nextSemanticToken()
 
 		case KeywordAccount:
 			access = ast.AccessAccount
+			// Skip the keyword
+			p.nextSemanticToken()
 
 		case KeywordContract:
 			access = ast.AccessContract
+			// Skip the keyword
+			p.nextSemanticToken()
 
 		case KeywordSelf:
 			access = ast.AccessPrivate
+			// Skip the keyword
+			p.nextSemanticToken()
 
 		default:
-			return ast.AccessNotSpecified, p.syntaxError(
-				"expected keyword %s, got %q",
-				enumeratedAccessModifierKeywords,
-				keyword,
-			)
-		}
+			firstTy, err := parseNominalType(p, lowestBindingPower, true)
```

Comment:
Maybe refactor the entitlement list parsing into a function (the code in this switch's default case)